test/e2e/adminactions_kubernetesobjects.go

```diff
@@ -0,0 +1,118 @@
+package e2e
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var _ = Describe("Admin actions", func() {
+	BeforeEach(runAdminTestsInDevOnly)
+
+	Specify("kubernetesobjects list", func() {
+		result, err := adminRequest("GET", "kubernetesobjects", "", nil, "kind=configmap", "namespace=openshift-machine-api")
+		Expect(err).NotTo(HaveOccurred())
+		obj := &unstructured.Unstructured{}
+		err = obj.UnmarshalJSON(result)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Build list of valid names from the kubeclient
+		configMaps, err := Clients.Kubernetes.CoreV1().ConfigMaps("openshift-machine-api").List(metav1.ListOptions{})
+		var validNames []string
+		for _, c := range configMaps.Items {
+			validNames = append(validNames, c.Name)
+		}
+
+		// Compare names from kubernetesobjects API with valid names from kubeclient
+		objs, err := obj.ToList()
+		Expect(err).NotTo(HaveOccurred())
+		for _, o := range objs.Items {
+			Expect(o.GetName()).To(BeElementOf(validNames))
+		}
```

Comment:
is there a risk that this could be flaky?
almost worth doing a single test as follows:
create object
get object
list (check object returned)
update object
delete object